File: exams/frmNationalExamPerformance.vb
Public Class frmNationalExamPerformance
    Private Function isvalid() As Boolean
        If cboYear.SelectedItem <> Nothing And cboExamination.SelectedItem <> Nothing Then
            Return True
        Else
            failure("Please Select An Examination To Analyze Results For!")
            Return False
        End If
    End Function
    Private Sub frmNationalExamPerformance_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        If Not connect() Then
            Me.Close()
        Else
            get_subjects()
            get_grades()
            For k As Integer = startyear To endyear
                cboYear.Items.Add(k)
            Next
            cboYear.SelectedItem = Today.Year
        End If
    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        If isvalid() Then
            exam_name = cboExamination.SelectedItem
            yr = cboYear.SelectedItem
            load_form()
            Dim frm As New frmNationalMeanAnalysis
            frm.ShowDialog()
        End If

    End Sub

    Private Sub Button3_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button3.Click
        If isvalid() Then
            exam_name = cboExamination.SelectedItem
            yr = cboYear.SelectedItem
            load_form()
            Dim frm As New frmNationalGradesAttained
            frm.ShowDialog()
        End If
    End Sub

    Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
        If isvalid() Then
            exam_name = cboExamination.SelectedItem
            yr = cboYear.SelectedItem
            load_form()
            rpt = "Subjects"
        End If
    End Sub

    Private Sub Button4_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button4.Click
        If isvalid() Then
            exam_name = cboExamination.SelectedItem
            yr = cboYear.SelectedItem
            load_form()
            rpt = "Students"
        End If
    End Sub

    Private Sub cboYear_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cboYear.SelectedIndexChanged
        If cboYear.SelectedItem.ToString <> None Then
            If stud Then
                query = "SELECT Name FROM national_examinations WHERE Year='" & cboYear.SelectedItem & "'"
            Else
                query = "SELECT Name FROM national_examinations WHERE Year='" & cboYear.SelectedItem & "'"
            End If
            cboExamination.Items.Clear()
            If qread(query) Then
                While dbreader.Read
                    cboExamination.Items.Add(dbreader("Name"))
                End While
            End If
        End If
    End Sub
End Class

File: exams/frmNationalExamPerformance.cs
using System;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace exams
{
    public partial class frmNationalExamPerformance
    {
        public frmNationalExamPerformance()
        {
            InitializeComponent();
            _cboYear.Name = "cboYear";
            _Button4.Name = "Button4";
            _Button3.Name = "Button3";
            _Button2.Name = "Button2";
            _Button1.Name = "Button1";
        }

        private bool isvalid()
        {
            if (Conversions.ToBoolean(Operators.AndObject(Operators.ConditionalCompareObjectNotEqual(cboYear.SelectedItem, null, false), Operators.ConditionalCompareObjectNotEqual(cboExamination.SelectedItem, null, false))))
            {
                return true;
            }
            else
            {
                publicSubsNFunctions.failure("Please Select An Examination To Analyze Results For!");
                return false;
            }
        }

        private void frmNationalExamPerformance_Load(object sender, EventArgs e)
        {
            if (!publicSubsNFunctions.connect())
            {
                Close();
            }
            else
            {
                publicSubsNFunctions.get_subjects();
                publicSubsNFunctions.get_grades();
                for (int k = publicSubsNFunctions.startyear, loopTo = publicSubsNFunctions.endyear; k <= loopTo; k++)
                    cboYear.Items.Add(k);
                cboYear.SelectedItem = DateAndTime.Today.Year;
            }
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            if (isvalid())
            {
                publicSubsNFunctions.exam_name = Conversions.ToString(cboExamination.SelectedItem);
                publicSubsNFunctions.yr = Conversions.ToInteger(cboYear.SelectedItem);
                publicSubsNFunctions.load_form();
                var frm = new frmNationalMeanAnalysis();
                frm.ShowDialog();
            }
        }

        private void Button3_Click(object sender, EventArgs e)
        {
            if (isvalid())
            {
                publicSubsNFunctions.exam_name = Conversions.ToString(cboExamination.SelectedItem);
                publicSubsNFunctions.yr = Conversions.ToInteger(cboYear.SelectedItem);
                publicSubsNFunctions.load_form();
                var frm = new frmNationalGradesAttained();
                frm.ShowDialog();
            }
        }

        private void Button2_Click(object sender, EventArgs e)
        {
            if (isvalid())
            {
                publicSubsNFunctions.exam_name = Conversions.ToString(cboExamination.SelectedItem);
                publicSubsNFunctions.yr = Conversions.ToInteger(cboYear.SelectedItem);
                publicSubsNFunctions.load_form();
                publicSubsNFunctions.rpt = "Subjects";
            }
        }

        private void Button4_Click(object sender, EventArgs e)
        {
            if (isvalid())
            {
                publicSubsNFunctions.exam_name = Conversions.ToString(cboExamination.SelectedItem);
                publicSubsNFunctions.yr = Conversions.ToInteger(cboYear.SelectedItem);
                publicSubsNFunctions.load_form();
                publicSubsNFunctions.rpt = "Students";
            }
        }

        private void cboYear_SelectedIndexChanged(object sender, EventArgs e)
        {
            if ((cboYear.SelectedItem.ToString() ?? "") != (publicSubsNFunctions.None ?? ""))
            {
                if (publicSubsNFunctions.stud)
                {
                    publicSubsNFunctions.query = Conversions.ToString(Operators.ConcatenateObject(Operators.ConcatenateObject("SELECT Name FROM national_examinations WHERE Year='", cboYear.SelectedItem), "'"));
                }
                else
                {
                    publicSubsNFunctions.query = Conversions.ToString(Operators.ConcatenateObject(Operators.ConcatenateObject("SELECT Name FROM national_examinations WHERE Year='", cboYear.SelectedItem), "'"));
                }

                cboExamination.Items.Clear();
                if (publicSubsNFunctions.qread(ref publicSubsNFunctions.query))
                {
                    while (publicSubsNFunctions.dbreader.Read())
                        cboExamination.Items.Add(publicSubsNFunctions.dbreader["Name"]);
                }
            }
        }
    }
}